File: julia/1_code/r_from_spherical.jl
using LegendrePolynomials

function r_from_spherical(θ::Float64, amplitudes::Vector{Float64})::Float64
    return sin(θ) * (1 + sum(amplitudes .* (collectPl(cos(θ), lmax = length(amplitudes)).parent)[2:end]));
end

File: matlab/1_code/r_from_spherical.m
function r = r_from_spherical(theta, amplitudes, PROBLEM_CONSTANTS)
    d = length(amplitudes);
    LP = PROBLEM_CONSTANTS.LEGENDRE_POLYNOMIALS;
    r = arrayfun(@(angle) ...
        sin(angle) * (1 + sum(dot(amplitudes, ...
        arrayfun(@(idx) LP{idx}(cos(angle)), 1:d)))), theta);
end